homeassistant/components/light/httplight.py

```diff
@@ -0,0 +1,147 @@
+"""
+Support for lights that use the HTTP protocol.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.pwm/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_TYPE
+from homeassistant.components.light import (
+    Light, ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR, SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR, PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.color as color_util
+
+
+REQUIREMENTS = ['requests==2.19.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TYPE_SIMPLE = 'simple'
+CONF_TYPE_DIMMABLE = 'dimmable'
+CONF_TYPE_COLOR = 'color'
+
+CONF_TYPES = [CONF_TYPE_SIMPLE, CONF_TYPE_DIMMABLE, CONF_TYPE_COLOR]
+
+SUPPORT_HTTPLIGHT_BRIGHTNESS = SUPPORT_BRIGHTNESS
+SUPPORT_HTTPLIGHT_COLOR = SUPPORT_BRIGHTNESS | SUPPORT_COLOR
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_TYPE): vol.In(CONF_TYPES)
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup HTTP Light platform for required type."""
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    light_type = config.get(CONF_TYPE)
+    add_devices([HTTPLight(host, name, light_type)])
+
+
+class HTTPLight(Light):
+    """Simple HTTP Light supports on and off."""
+
+    def __init__(self, host, name, light_type):
+        """Initialize the light."""
+        self._name = name
+        self._host = host
+        self._type = light_type
+        self._brightness = 0
+        self._hs_color = [0, 0]
+        self._state = False
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def hs_color(self):
+        """Return the color of the light."""
+        return self._hs_color
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._type == CONF_TYPE_DIMMABLE:
+            return SUPPORT_HTTPLIGHT_BRIGHTNESS
+        elif self._type == CONF_TYPE_COLOR:
+            return SUPPORT_HTTPLIGHT_COLOR
```

Comment:
Should color support setting brightness? It looks like it should.

Reply:
It does. See the definition for `SUPPORT_HTTPLIGHT_COLOR`.